app/models/miq_expression.rb

```diff
@@ -1496,7 +1496,12 @@ def self.build_relats(model, parent = {}, seen = [])
     result = {:columns => model.column_names_with_virtual, :parent => parent}
     result[:reflections] = {}
 
-    model.reflections_with_virtual.each do |assoc, ref|
+    refs = model.reflections_with_virtual
+    model.descendants.each do |desc|
```

Comment:
This is unfortunate, because it forces us to load all the providers.

Of course, there's a good chance we already have, for some other reason... but adding more only cements that situation.

That said, I don't have an alternative suggestion. ¯\\\_(ツ)\_/¯